PICARD-1630: Always save FLAC picture blocks after Vorbis comments

phw opened the pull request:
<!--
    Hello! Thanks for submitting a pull request to MusicBrainz Picard. We
    appreciate your time and interest in helping our project!

    Use this template to help us review your change. Not everything is required,
    depending on your change. Keep or delete what is relevant for your change.
    Remember that it helps us review if you give more helpful info for us to
    understand your change.

    Ensure that you've read through and followed the Contributing Guidelines, in
    [CONTRIBUTING.md](https://github.com/metabrainz/picard/blob/master/CONTRIBUTING.md).
-->

# Summary
If FLAC picture blocks are located before the Vorbis comments, Windows cannot read the tags. By ensuring the pictures are always located after the Vorbis comments the FLAC metadata will be visible and editable in Windows Explorer.
<!--
    Update the checkbox with an [x] for the type of contribution you are making.
-->

* This is a…
    * [x] Bug fix
    * [ ] Feature addition
    * [ ] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

<!--
    Anything that helps us understand why you are making this change goes here.
    What problem are you trying to fix? What does this change address?
-->

* JIRA ticket (_optional_): [PICARD-1630](https://tickets.metabrainz.org/browse/PICARD-1630)
<!--
    Please make sure you prefix your pull request title with 'PICARD-XXX' in order
    for our ticket tracker to link your pull request to the relevant ticket.
-->



# Solution
Always sort the FLAC metadata blocks so picture blocks are located after the Vorbis comments block. This does not affect the order of any other blocks, also the order of the picture blocks is maintained.
<!--
    The details of your change. Talk about technical details, considerations, or
    other interesting points. If you have a lot to say, be more detailed in this
    section.
-->


# Action

<!--
    Other than merging your change, do you want / need us to do anything else
    with your change? This could include reviewing a specific part of your PR.
-->

zas commented on picard/formats/vorbis.py:
```diff
@@ -69,6 +69,27 @@ def is_valid_key(key):
+        # No tags found, nothing to sort
+        return
+
+    # Now move those picture block after the tag block, maintaining their order.
```
> maintaining their order ?

Perhaps I'm missing something here.
List of pictures is built from previous loop, using `append()`, but `insert(index, value)` will actually revere the order:

```python
>>> l = ['a', 'b', 'c']
>>> l
['a', 'b', 'c']
>>> for p in ['pic1', 'pic2', 'pic3']:
...     l.insert(1, p)
... 
>>> l
['a', 'pic3', 'pic2', 'pic1', 'b', 'c']
```

phw replied:
Ha, my first thinking was also that I would need to insert at tagindex + 1 and then increment. But turns out the insert position is stable at tagindex, because one by one the blocks are removedat the start and then inserted again. The removal automatically moves all following elements, making room for the insertion.

phw replied:
Just to demonstrate what I mean with your example:

```python
>>> l = ['pic1', 'pic2', 'pic3', 'a', 'b', 'c']
>>> l
['pic1', 'pic2', 'pic3', 'a', 'b', 'c']
>>> for p in ['pic1', 'pic2', 'pic3']:
...     l.remove(p)
...     l.insert(3, p)
...
>>> l
['a', 'pic1', 'pic2', 'pic3', 'b', 'c']

```

zas replied:
I got it, missed this remove part ;)

phw commented on picard/formats/vorbis.py:
```diff
@@ -69,6 +69,27 @@ def is_valid_key(key):
+    picblocks = []
+    for b in metadata_blocks:
+        if b.code == mutagen.flac.VCFLACDict.code:
+            tagblock = b
```
Oh yes, that's left over code from previous iteration. Will fix it

phw commented:
@zas I addressed your comments, please check.

> I'm not fond of fixing windows... but ok.

Me neither, but the majority of our user base is there. Also having (often multiple MB large) picture blocks after the tags isn't the worst of design decisions, this makes the metadata itself available earlier. Even foobar2000 decided of going this route, and they otherwise seem rather reluctant to implement anything too specific to other applications.

phw commented:
@zas I added some documentation to `flac_sort_pics_after_tags` to make the purpose of this clear. Otherwise this is unchanged.
